src/userspace/exec_monitor.cpp

```diff
@@ -65,11 +66,31 @@ int ExecMonitor::run()
 		std::cerr << "[EXEC_MONITOR]: Error allocating the ringbufffer.\n";
 		goto out;
 	}
-	
+
+	if (n_proc > 0) {
+		skel->bss->n_monitored_proc = 0;
+		skel->bss->max_n_proc = n_proc;
+	}
+
+	if (ppid_list.size() != 0) {
+		skel->bss->filter_by_ppid = 1;
+		int list_size = 10;
+		if (ppid_list.size() < 10) {
+			list_size = ppid_list.size();
+		}
```

Comment:
This could be done faster with something like `int list_size = min(ppid_list.size(), PPID_LIST_MAP_LEN);`

http://www.cplusplus.com/reference/algorithm/min/